internal/server/routes.go

```diff
@@ -56,7 +57,14 @@ func (s *Server) GenerateRoutes() Routes {
 
 	get(a, authn, "/api/users", a.ListUsers)
 	post(a, authn, "/api/users", a.CreateUser)
-	get(a, authn, "/api/users/:id", a.GetUser)
+	// getuser needs its own route declaration because it may call database write operations on sync
+	getUserRoute := route[api.GetUserRequest, *api.User]{
+		handler: a.GetUser,
+		routeSettings: routeSettings{
+			omitFromTelemetry: true,
+		},
```

Comment:
I think this is a sign we might not have the right design for this.  

If we need to write during a `GET` request can we do that in the middleware before handing off the read-only transaction to the handler?

Is there something special about `GetUser`? Why do it in that endpoint instead of approval of the device flow (which is already a POST) ?

Could we do this in the background in the middleware? That would also fix the problem of latency communicating with the provider causing slow API requests.

Reply:
I did it here because its the endpoint that the UI uses to check session validity. Moved to the middleware instead from your feedback.